Doc/library/intro.rst

```diff
@@ -64,3 +64,50 @@ Notes on availability
   libc version, then both conditions must hold. For example a feature with note
   *Availability: Linux >= 3.17 with glibc >= 2.27* requires both Linux 3.17 or
   newer and glibc 2.27 or newer.
+
+.. _wasm-availability:
+
+Notes on WebAssembly platforms
+==============================
+
+The WebAssembly platforms ``wasm32-emscripten`` and ``wasm32-wasi`` provide a
+subset of POSIX APIs. WASM runtimes and browser are sandboxed and have
+limited access to the host and external resources. Any standard library module
+that uses processes, threading, networking (:mod:`socket`), signals, or
+other forms of inter-process communication (IPC), is either not available
+or may not work as on other Unix-like systems. File I/O, file system, and Unix
+permission-related functions are restricted, too. Emscripten does not permit
+blocking operations like blocking network I/O and :func:`~time.sleep`.
+
+The properties and behavior of Python on WebAssembly platforms depend on the
+`Emscripten`_ or `WASI`_-SDK version, WASM runtimes (browser, NodeJS, `wasmtime`_),
+and Python build time flags. WebAssembly, Emscripten, and WASI are evolving.
+Some features like networking may be supported in the future.
```

Comment:
Given that Emscripten and WASI runtimes are moving targets, there could at least be unofficial builds of Python build against future versions of emscripten where some of the features marked as unavailable become available.  Do we want to somehow flag the difference between "explicitly disabled in Python" vs. "not known to work in current environments"?  *Maybe* we want to add to this paragraph:

"Python features documented as unsupported may in fact work depending on your environment."

But *maybe* that's too confusing for most users to understand, and something we can address in the future as these things mature.

Reply:
Very few features are explicitly disabled. The ``config.site`` files disable a few syscalls that fail tests or make no sense. For example I disabled the effective user/group id function family. Emscripten returns 0, which some code interprets as `CAP_SYS_ADMIN` privileges. The `config.site` files in `Tool/wasm` directory have comments for special cases.

The target audience for the section is "curious users without previous WASM experience". I want to give them *some* ideas what does not work without going into too many technical details.